diff -urN a/PyIex/Makefile.am b/PyIex/Makefile.am
--- a/PyIex/Makefile.am	2017-11-17 23:31:37.000000000 +0000
+++ b/PyIex/Makefile.am	2018-01-26 11:39:32.346291932 +0000
@@ -1,21 +1,8 @@
 ## Process this file with automake to produce Makefile.in
 
-pyexec_LTLIBRARIES = iexmodule.la
-lib_LTLIBRARIES = libPyIex.la
-
-libPyIex_la_SOURCES = PyIex.cpp 
-
 libPyIexinclude_HEADERS = PyIex.h PyIexExport.h PyIexTypeTranslator.h
-
-libPyIex_la_LDFLAGS = -version-info @LIBTOOL_VERSION@ \
-                      -no-undefined
-libPyIex_la_LIBADD  = -lz @ILMBASE_LIBS@ @BOOST_PYTHON_LIBS@
 libPyIexincludedir  = $(includedir)/OpenEXR
 
-iexmodule_la_SOURCES = iexmodule.cpp
-iexmodule_la_LDFLAGS = -avoid-version -module
-iexmodule_la_LIBADD  = libPyIex.la @BOOST_PYTHON_LIBS@
-
 noinst_HEADERS = 
 
 INCLUDES = @ILMBASE_CXXFLAGS@ \
diff -urN a/PyImath/Makefile.am b/PyImath/Makefile.am
--- a/PyImath/Makefile.am	2017-11-17 23:31:37.000000000 +0000
+++ b/PyImath/Makefile.am	2018-01-26 11:40:24.880276439 +0000
@@ -1,39 +1,5 @@
 ## Process this file with automake to produce Makefile.in
 
-pyexec_LTLIBRARIES = imathmodule.la
-lib_LTLIBRARIES = libPyImath.la
-
-libPyImath_la_SOURCES = PyImath.cpp \
-    PyImathAutovectorize.cpp \
-    PyImathBox2Array.cpp \
-    PyImathBox3Array.cpp \
-    PyImathBox.cpp \
-    PyImathColor3.cpp \
-    PyImathColor4.cpp \
-    PyImathEuler.cpp \
-    PyImathFixedArray.cpp \
-    PyImathFrustum.cpp \
-    PyImathLine.cpp \
-    PyImathMatrix33.cpp \
-    PyImathMatrix44.cpp \
-    PyImathPlane.cpp \
-    PyImathQuat.cpp \
-    PyImathRandom.cpp \
-    PyImathShear.cpp \
-    PyImathStringArray.cpp \
-    PyImathStringTable.cpp \
-    PyImathTask.cpp \
-    PyImathUtil.cpp \
-    PyImathFixedVArray.cpp \
-    PyImathVec2fd.cpp \
-    PyImathVec2si.cpp \
-    PyImathVec3fd.cpp \
-    PyImathVec3siArray.cpp \
-    PyImathVec3si.cpp \
-    PyImathVec4fd.cpp \
-    PyImathVec4siArray.cpp \
-    PyImathVec4si.cpp
-
 libPyImathinclude_HEADERS = PyImath.h \
     PyImathAutovectorize.h \
     PyImathBoxArrayImpl.h \
@@ -71,19 +37,8 @@
     PyImathVec.h \
     PyImathVecOperators.h
 
-libPyImath_la_LDFLAGS = -version-info @LIBTOOL_VERSION@ \
-                        -no-undefined
-libPyImath_la_LIBADD  = -lz $(top_builddir)/PyIex/libPyIex.la \
-                        @ILMBASE_LIBS@ @BOOST_PYTHON_LIBS@
 libPyImathincludedir  = $(includedir)/OpenEXR
 
-imathmodule_la_SOURCES = imathmodule.cpp \
-    PyImathFun.cpp \
-    PyImathBasicTypes.cpp 
-
-imathmodule_la_LDFLAGS = -avoid-version -module
-imathmodule_la_LIBADD  = -lPyImath @BOOST_PYTHON_LIBS@
-
 noinst_HEADERS = PyImathFun.h \
     PyImathBasicTypes.h 
 
diff -urN a/PyImathNumpy/Makefile.am b/PyImathNumpy/Makefile.am
--- a/PyImathNumpy/Makefile.am	2017-11-17 23:31:37.000000000 +0000
+++ b/PyImathNumpy/Makefile.am	2018-01-26 11:40:42.774271162 +0000
@@ -1,11 +1,5 @@
 ## Process this file with automake to produce Makefile.in
 
-pyexec_LTLIBRARIES = imathnumpymodule.la
-
-imathnumpymodule_la_SOURCES = imathnumpymodule.cpp
-imathnumpymodule_la_LDFLAGS = -avoid-version -module
-imathnumpymodule_la_LIBADD  = $(top_builddir)/PyImath/libPyImath.la @BOOST_PYTHON_LIBS@
-
 noinst_HEADERS = 
 
 INCLUDES = @ILMBASE_CXXFLAGS@      \
