diff -Naur a/intern/cycles/CMakeLists.txt b/intern/cycles/CMakeLists.txt
--- a/intern/cycles/CMakeLists.txt	2016-06-16 18:11:45.083958037 +0930
+++ b/intern/cycles/CMakeLists.txt	2016-06-16 18:16:01.864954315 +0930
@@ -138,8 +138,12 @@
 	)
 endif()
 
-set(WITH_CYCLES_DEVICE_OPENCL TRUE)
-set(WITH_CYCLES_DEVICE_CUDA TRUE)
+if(WITH_OPENCL)
+	set(WITH_CYCLES_DEVICE_OPENCL TRUE)
+endif()
+if(WITH_CUDA)
+	set(WITH_CYCLES_DEVICE_CUDA TRUE)
+endif()
 set(WITH_CYCLES_DEVICE_MULTI TRUE)
 
 if(CYCLES_STANDALONE_REPOSITORY)
