diff --git a/Half/CMakeLists.txt b/Half/CMakeLists.txt
index 6f9714d..958d1b0 100644
--- a/Half/CMakeLists.txt
+++ b/Half/CMakeLists.txt
@@ -1,23 +1,24 @@
 # yue.nicholas@gmail.com
 
 ADD_EXECUTABLE ( eLut eLut.cpp )
-
-ADD_CUSTOM_COMMAND (
-  TARGET eLut POST_BUILD
-  COMMAND eLut > ${CMAKE_CURRENT_BINARY_DIR}/eLut.h
+ADD_CUSTOM_COMMAND(
+  OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/eLut.h
+  COMMAND eLut ARGS > ${CMAKE_CURRENT_BINARY_DIR}/eLut.h
+  DEPENDS eLut
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-)
+  )
 SET_SOURCE_FILES_PROPERTIES(
   ${CMAKE_CURRENT_BINARY_DIR}/eLut.h
   PROPERTIES HEADER_FILE_ONLY TRUE
   )
 
 ADD_EXECUTABLE ( toFloat toFloat.cpp )
-ADD_CUSTOM_COMMAND (
-  TARGET toFloat POST_BUILD
-  COMMAND toFloat > ${CMAKE_CURRENT_BINARY_DIR}/toFloat.h
+ADD_CUSTOM_COMMAND(
+  OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/toFloat.h
+  COMMAND toFloat ARGS > ${CMAKE_CURRENT_BINARY_DIR}/toFloat.h
+  DEPENDS toFloat
   WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}
-)
+  )
 SET_SOURCE_FILES_PROPERTIES(
   ${CMAKE_CURRENT_BINARY_DIR}/toFloat.h
   PROPERTIES HEADER_FILE_ONLY TRUE
@@ -27,9 +28,7 @@ SET_SOURCE_FILES_PROPERTIES(
   half.cpp
   PROPERTIES
   OBJECT_DEPENDS
-  ${CMAKE_CURRENT_BINARY_DIR}/eLut.h
-  OBJECT_DEPENDS
-  ${CMAKE_CURRENT_BINARY_DIR}/toFloat.h
+  "${CMAKE_CURRENT_BINARY_DIR}/eLut.h;${CMAKE_CURRENT_BINARY_DIR}/toFloat.h"
   )
 
 IF(BUILD_SHARED_LIBS)
diff --git a/Half/half.cpp b/Half/half.cpp
index 609aaa1..09a50aa 100644
--- a/Half/half.cpp
+++ b/Half/half.cpp
@@ -94,9 +94,9 @@ half::convert (int i)
     // of float and half (127 versus 15).
     //
 
-    register int s =  (i >> 16) & 0x00008000;
-    register int e = ((i >> 23) & 0x000000ff) - (127 - 15);
-    register int m =   i        & 0x007fffff;
+    int s =  (i >> 16) & 0x00008000;
+    int e = ((i >> 23) & 0x000000ff) - (127 - 15);
+    int m =   i        & 0x007fffff;
 
     //
     // Now reassemble s, e and m into a half:
diff --git a/Half/half.h b/Half/half.h
index f78e4f6..da418d6 100644
--- a/Half/half.h
+++ b/Half/half.h
@@ -459,7 +459,7 @@ half::half (float f)
 	// to do the float-to-half conversion.
 	//
 
-	register int e = (x.i >> 23) & 0x000001ff;
+	int e = (x.i >> 23) & 0x000001ff;
 
 	e = _eLut[e];
 
@@ -470,7 +470,7 @@ half::half (float f)
 	    // bits and combine it with the sign and exponent.
 	    //
 
-	    register int m = x.i & 0x007fffff;
+	    int m = x.i & 0x007fffff;
 	    _h = e + ((m + 0x00000fff + ((m >> 13) & 1)) >> 13);
 	}
 	else
diff --git a/Imath/ImathMatrix.h b/Imath/ImathMatrix.h
index 3e96c2f..1d7f8dc 100644
--- a/Imath/ImathMatrix.h
+++ b/Imath/ImathMatrix.h
@@ -2527,11 +2527,11 @@ Matrix44<T>::multiply (const Matrix44<T> &a,
                        const Matrix44<T> &b,
                        Matrix44<T> &c)
 {
-    register const T * IMATH_RESTRICT ap = &a.x[0][0];
-    register const T * IMATH_RESTRICT bp = &b.x[0][0];
-    register       T * IMATH_RESTRICT cp = &c.x[0][0];
+    const T * IMATH_RESTRICT ap = &a.x[0][0];
+    const T * IMATH_RESTRICT bp = &b.x[0][0];
+          T * IMATH_RESTRICT cp = &c.x[0][0];
 
-    register T a0, a1, a2, a3;
+    T a0, a1, a2, a3;
 
     a0 = ap[0];
     a1 = ap[1];
diff --git a/Makefile.am b/Makefile.am
index a2e5370..b3c4e86 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -15,7 +15,7 @@ DIST_SUBDIRS = \
 
 EXTRA_DIST = \
 	AUTHORS COPYING ChangeLog INSTALL LICENSE NEWS \
-	README README.CVS README.OSX README.win32 \
+	README README.CVS README.OSX \
 	README.cmake.txt README.namespacing README.git \
 	bootstrap \
 	config.windows/IlmBaseConfig.h \
