diff -Naur a/ui.py b/ui.py
--- a/ui.py	2016-11-15 23:43:25.000000000 +1030
+++ b/ui.py	2016-12-07 02:24:26.181058470 +1030
@@ -435,7 +435,7 @@
 
         row = col.row()
         row.prop(rm, "show_integrator_settings", icon=icon, text=text,
-                 icon_only=True, emboss=False)
+                 icon_only=False, emboss=False)
         if rm.show_integrator_settings:
             draw_props(integrator_settings,
                        integrator_settings.prop_names, col)
