diff -Naur a/opensubdiv/CMakeLists.txt b/opensubdiv/CMakeLists.txt
--- a/opensubdiv/CMakeLists.txt	2016-05-30 23:30:07.196057673 +0930
+++ b/opensubdiv/CMakeLists.txt	2016-05-30 23:32:01.721049102 +0930
@@ -106,7 +106,7 @@
     if( CUDA_FOUND )
         include_directories( "${CUDA_INCLUDE_DIRS}" )
         if (UNIX)
-            list( APPEND CUDA_NVCC_FLAGS -Xcompiler -fPIC --gpu-architecture compute_11 )
+            list( APPEND CUDA_NVCC_FLAGS -Xcompiler -fPIC --gpu-architecture compute_20 )
         endif()
     endif()
 
diff -Naur a/opensubdiv/osd/CMakeLists.txt b/opensubdiv/osd/CMakeLists.txt
--- a/opensubdiv/osd/CMakeLists.txt	2016-05-30 23:31:01.981053573 +0930
+++ b/opensubdiv/osd/CMakeLists.txt	2016-05-30 23:30:09.242057520 +0930
@@ -292,7 +292,7 @@
     endif()
 
     if (UNIX)
-        list( APPEND CUDA_NVCC_FLAGS -Xcompiler -fPIC --gpu-architecture compute_11 )
+        list( APPEND CUDA_NVCC_FLAGS -Xcompiler -fPIC --gpu-architecture compute_20 )
     endif()
 endif()
 
