From 2eb197d8b82e29fdaf65a7cf53452e401afecd33 Mon Sep 17 00:00:00 2001
From: Jonathan Scruggs <j.scruggs@gmail.com>
Date: Wed, 18 Oct 2017 10:18:25 +0100
Subject: [PATCH] openexr_viewers: configure.ac fixes

---
 configure.ac | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure.ac b/configure.ac
index 94c88cd..da58f33 100644
--- a/configure.ac
+++ b/configure.ac
@@ -146,10 +146,10 @@ AM_COMPILELINKRUN(
     FrameBuffer outFb;
 
     ImfCtl::applyTransforms (interpreter, transformNames, transformWindow,
-                             envHeader, inHeader, inFb, outHeader, outFb);
+                             envHeader, inHeader, inFb, outHeader, outFb, 1);
    ]],
    AC_MSG_RESULT([Compiled and ran OpenEXR_CTL test program.]); have_ctl=yes, 
-   AC_MSG_RESULT([The OpenEXR_CTL package could not be found.  Will build and install playexr and exrdisplay without ctl support. ]); have_ctl=no)
+   AC_MSG_ERROR([The OpenEXR_CTL package could not be found.  Will build and install playexr and exrdisplay without ctl support. ]); have_ctl=no)
 
 
 
-- 
2.14.2

