diff --git a/python/pyOpenVDBModule.cc b/python/pyOpenVDBModule.cc
index 508c787..5796134 100644
--- a/python/pyOpenVDBModule.cc
+++ b/python/pyOpenVDBModule.cc
@@ -583,6 +583,16 @@ struct VecTypeDescr
 #define PY_OPENVDB_MODULE_NAME  pyopenvdb
 #endif
 
+#if PY_MAJOR_VERSION >= 3
+int
+#else
+void
+#endif
+init_numpy()
+{
+    import_array();
+}
+
 BOOST_PYTHON_MODULE(PY_OPENVDB_MODULE_NAME)
 {
     // Don't auto-generate ugly, C++-style function signatures.
@@ -592,7 +602,7 @@ BOOST_PYTHON_MODULE(PY_OPENVDB_MODULE_NAME)
 
 #ifdef PY_OPENVDB_USE_NUMPY
     // Initialize NumPy.
-    import_array();
+    init_numpy();
 #endif
 
     using namespace openvdb::OPENVDB_VERSION_NAME;
