diff -Naur a/Makefile b/Makefile
--- a/Makefile	2016-06-07 23:30:03.586728379 +0930
+++ b/Makefile	2016-06-07 23:29:52.015729126 +0930
@@ -907,6 +907,14 @@
 	    cp -r -f doc/html $(DESTDIR)/share/doc/openvdb; \
 	    echo "Copied documentation to $(DESTDIR)/share/doc/openvdb/"; \
 	fi
+	@#
+	if [ -d doc/latex ]; \
+	then \
+	    mkdir -p $(DESTDIR)/share/doc/openvdb; \
+	    echo "Created $(DESTDIR)/share/doc/openvdb/"; \
+	    cp -r -f doc/latex $(DESTDIR)/share/doc/openvdb; \
+	    echo "Copied documentation to $(DESTDIR)/share/doc/openvdb/"; \
+	fi
 
 # TODO: This accumulates all source file dependencies into a single file
 # containing a rule for each *.o file.  Consider generating a separate
