diff -purN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	2016-09-24 17:55:23.000000000 +0100
+++ b/CMakeLists.txt	2016-10-29 15:01:07.686134834 +0100
@@ -89,24 +89,26 @@ if(NOT HAVE_B64_IN_LIBC)
     endif(HAVE_B64_IN_LIBRESOLV)
 endif(NOT HAVE_B64_IN_LIBC)
 
-if(Threads_FOUND)
+if(WITH_THREADS)
   option(TIGHTVNC_FILETRANSFER "Enable filetransfer" ON)
-endif(Threads_FOUND)
-if(ZLIB_FOUND)
+endif(WITH_THREADS)
+if(WITH_ZLIB)
   set(LIBVNCSERVER_HAVE_LIBZ 1)
-endif(ZLIB_FOUND)
-if(JPEG_FOUND)
+endif(WITH_ZLIB)
+if(WITH_JPEG)
   set(LIBVNCSERVER_HAVE_LIBJPEG 1)
-endif(JPEG_FOUND)
-if(PNG_FOUND)
+endif(WITH_JPEG)
+if(WITH_PNG)
   set(LIBVNCSERVER_HAVE_LIBPNG 1)
-endif(PNG_FOUND)
-option(LIBVNCSERVER_ALLOW24BPP "Allow 24 bpp" ON)
+endif(WITH_PNG)
+if(WITH_24BPP)
+	option(LIBVNCSERVER_ALLOW24BPP "Allow 24 bpp" ON)
+endif(WITH_24BPP)
 
 pkg_check_modules(SYSTEMD "libsystemd")
-if(SYSTEMD_FOUND)
+if(WITH_SYSTEMD)
   option(LIBVNCSERVER_WITH_SYSTEMD "Build with systemd socket activation support" ON)
-endif(SYSTEMD_FOUND)
+endif(WITH_SYSTEMD)
 
 if(LIBVNCSERVER_WITH_SYSTEMD)
   add_definitions(-DLIBVNCSERVER_WITH_SYSTEMD)
@@ -114,13 +116,13 @@ if(LIBVNCSERVER_WITH_SYSTEMD)
   set(ADDITIONAL_LIBS ${ADDITIONAL_LIBS} ${SYSTEMD_LIBRARIES})
 endif(LIBVNCSERVER_WITH_SYSTEMD)
 
-if(GNUTLS_FOUND)
+if(WITH_GNUTLS)
   set(LIBVNCSERVER_WITH_CLIENT_TLS 1)
   option(LIBVNCSERVER_WITH_WEBSOCKETS "Build with websockets support (gnutls)" ON)
   set(WEBSOCKET_LIBRARIES ${RESOLV_LIB} ${GNUTLS_LIBRARIES})
   set(WSSRCS ${LIBVNCSERVER_DIR}/rfbssl_gnutls ${LIBVNCSERVER_DIR}/rfbcrypto_gnutls)
   include_directories(${GNUTLS_INCLUDE_DIR})
-elseif(OPENSSL_FOUND)
+elseif(WITH_SSL)
   option(LIBVNCSERVER_WITH_WEBSOCKETS "Build with websockets support (openssl)" ON)
   set(WEBSOCKET_LIBRARIES ${RESOLV_LIB} ${OPENSSL_LIBRARIES})
   set(WSSRCS ${LIBVNCSERVER_DIR}/rfbssl_openssl ${LIBVNCSERVER_DIR}/rfbcrypto_openssl)
@@ -131,11 +133,11 @@ else()
   set(WSSRCS ${LIBVNCSERVER_DIR}/rfbssl_none.c ${LIBVNCSERVER_DIR}/rfbcrypto_included.c ${COMMON_DIR}/md5.c ${COMMON_DIR}/sha1.c)
 endif()
 
-if(LIBGCRYPT_LIBRARIES)
+if(WITH_GCRYPT AND LIBGCRYPT_LIBRARIES)
   message(STATUS "Found libgcrypt: ${LIBGCRYPT_LIBRARIES}")
   set(LIBVNCSERVER_WITH_CLIENT_GCRYPT 1)
   set(ADDITIONAL_LIBS ${ADDITIONAL_LIBS} ${LIBGCRYPT_LIBRARIES})
-endif(LIBGCRYPT_LIBRARIES)
+endif(WITH_GCRYPT AND LIBGCRYPT_LIBRARIES)
 
 
 check_include_file("endian.h"      LIBVNCSERVER_HAVE_ENDIAN_H)
@@ -162,7 +164,7 @@ check_function_exists(vfork           LI
 check_function_exists(vprintf         LIBVNCSERVER_HAVE_VPRINTF)
 
 
-if(LIBVNCSERVER_HAVE_WS2TCPIP_H AND LIBVNCSERVER_HAVE_VPRINTF)
+if(LIBVNCSERVER_HAVE_WS2TCPIP_H AND LIBVNCSERVER_HAVE_VPRINTF AND WITH_IPV6)
 	option(LIBVNCSERVER_IPv6 "Enable IPv6 Support" ON)
 endif()
 
@@ -228,12 +230,12 @@ set(LIBVNCCLIENT_SOURCES
     ${COMMON_DIR}/minilzo.c
 )
 
-if(GNUTLS_FOUND)
+if(WITH_GNUTLS)
   set(LIBVNCCLIENT_SOURCES
     ${LIBVNCCLIENT_SOURCES}
     ${LIBVNCCLIENT_DIR}/tls_gnutls.c
   )
-elseif(OPENSSL_FOUND)
+elseif(WITH_SSL)
   set(LIBVNCCLIENT_SOURCES
     ${LIBVNCCLIENT_SOURCES}
     ${LIBVNCCLIENT_DIR}/tls_openssl.c
@@ -245,7 +247,7 @@ else()
   )
 endif()
 
-if(ZLIB_FOUND)
+if(WITH_ZLIB)
   add_definitions(-DLIBVNCSERVER_HAVE_LIBZ)
   include_directories(${ZLIB_INCLUDE_DIR})
   set(LIBVNCSERVER_SOURCES
@@ -255,19 +257,19 @@ if(ZLIB_FOUND)
     ${LIBVNCSERVER_DIR}/zrleoutstream.c
     ${LIBVNCSERVER_DIR}/zrlepalettehelper.c
   )
-endif(ZLIB_FOUND)
+endif(WITH_ZLIB)
 
-if(JPEG_FOUND)
+if(WITH_JPEG)
   add_definitions(-DLIBVNCSERVER_HAVE_LIBJPEG)
   include_directories(${JPEG_INCLUDE_DIR})
   set(TIGHT_C ${LIBVNCSERVER_DIR}/tight.c ${COMMON_DIR}/turbojpeg.c)
-endif(JPEG_FOUND)
+endif(WITH_JPEG)
 
-if(PNG_FOUND)
+if(WITH_PNG)
   add_definitions(-DLIBVNCSERVER_HAVE_LIBPNG)
   include_directories(${PNG_INCLUDE_DIR})
   set(TIGHT_C ${LIBVNCSERVER_DIR}/tight.c ${COMMON_DIR}/turbojpeg.c)
-endif(PNG_FOUND)
+endif(WITH_PNG)
 
 set(LIBVNCSERVER_SOURCES
     ${LIBVNCSERVER_SOURCES}
@@ -336,12 +338,12 @@ set(LIBVNCSERVER_TESTS
     vncev
   )
 
-if(Threads_FOUND)
+if(WITH_THREADS)
   set(LIBVNCSERVER_TESTS
     ${LIBVNCSERVER_TESTS}
     blooptest
   )
-endif(Threads_FOUND)
+endif(WITH_THREADS)
 
 if(TIGHTVNC_FILETRANSFER)
   set(LIBVNCSERVER_TESTS
@@ -362,14 +364,14 @@ set(LIBVNCCLIENT_TESTS
     ppmtest
 )
 
-if(SDL_FOUND)
+if(WITH_SDL)
   include_directories(${SDL_INCLUDE_DIR})
   set(LIBVNCCLIENT_TESTS
     ${LIBVNCCLIENT_TESTS}
     SDLvncviewer
   )
   set(SDLvncviewer_EXTRA_SOURCES scrap.c)
-endif(SDL_FOUND)
+endif(WITH_SDL)
 
 if(HAVE_FFMPEG)
   set(LIBVNCCLIENT_TESTS
@@ -380,16 +382,20 @@ endif(HAVE_FFMPEG)
 
 
 file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/examples)
-foreach(test ${LIBVNCSERVER_TESTS})
-  add_executable(examples_${test} ${LIBVNCSRVTEST_DIR}/${test}.c)
-  target_link_libraries(examples_${test} vncserver ${CMAKE_THREAD_LIBS_INIT})
-endforeach(test ${LIBVNCSERVER_TESTS})
+if(WITH_TESTS)
+  foreach(test ${LIBVNCSERVER_TESTS})
+    add_executable(examples_${test} ${LIBVNCSRVTEST_DIR}/${test}.c)
+    target_link_libraries(examples_${test} vncserver ${CMAKE_THREAD_LIBS_INIT})
+  endforeach(test ${LIBVNCSERVER_TESTS})
+endif(WITH_TESTS)
 
 file(MAKE_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/client_examples)
-foreach(test ${LIBVNCCLIENT_TESTS})
-  add_executable(client_examples_${test} ${LIBVNCCLITEST_DIR}/${test}.c ${LIBVNCCLITEST_DIR}/${${test}_EXTRA_SOURCES} )
-  target_link_libraries(client_examples_${test} vncclient ${CMAKE_THREAD_LIBS_INIT} ${X11_LIBRARIES} ${SDL_LIBRARY} ${FFMPEG_LIBRARIES})
-endforeach(test ${LIBVNCCLIENT_TESTS})
+if(WITH_TESTS)
+  foreach(test ${LIBVNCCLIENT_TESTS})
+    add_executable(client_examples_${test} ${LIBVNCCLITEST_DIR}/${test}.c ${LIBVNCCLITEST_DIR}/${${test}_EXTRA_SOURCES} )
+    target_link_libraries(client_examples_${test} vncclient ${CMAKE_THREAD_LIBS_INIT} ${X11_LIBRARIES} ${SDL_LIBRARY} ${FFMPEG_LIBRARIES})
+  endforeach(test ${LIBVNCCLIENT_TESTS})
+endif(WITH_TESTS)
 
 install_targets(/lib vncserver)
 install_targets(/lib vncclient)
