diff -purN a/configure b/configure
--- a/configure	2016-03-28 15:51:58.000000000 +0100
+++ b/configure	2016-09-12 14:26:21.506308375 +0100
@@ -8716,10 +8716,8 @@ done
 test -n "$latex" || latex="no"
 
 export latex;
-if test $latex = "no" ;
+if test $latex != "no" ;
 then
-        as_fn_error $? "Unable to find a LaTeX application" "$LINENO" 5;
-fi
 
 
 
@@ -10503,6 +10501,7 @@ $as_echo "$as_me: WARNING: missing LaTeX
 
 
 fi
+fi # if test $latex = "no"
 
 if test "$PERL" != false; then
 
